### PR TITLE
Bump libraries and apply ksp on dagger-hilt, room

### PR DESCRIPTION
**Pull request by l2hyunwoo**
[Significant Changes]

- Change compile, target sdk version: 33 -> 34 to apply ksp dagger and room
- Change Compose Compiler Version: 1.4.3 -> 1.5.1
- Compose BOM: 06.00 -> 09.00
- Kotlin: 1.8.10 -> 1.9.10
- Android Gradle Plugin: 1.8.10 -> 1.9.10
- Dagger Hilt: 2.46.1 -> 2.48
- Apply ksp on ksp, room

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -16,8 +20,8 @@ dependencies {
     implementation(libs.bundles.retrofit)
     implementation(libs.timber)
     implementation(libs.bundles.room)
-    kapt(libs.room.compiler)
+    ksp(libs.room.compiler)
```
Comment (Nanamare) [inline app/build.gradle.kts:23]: @l2hyunwoo 🌟

File: app/src/main/java/com/nanamare/movie/ui/screen/TrendingScreen.kt [hunk 2]
```diff
@@ -59,9 +58,9 @@ fun TrendingList(
     block: (NavigationViewModel.Screen) -> Unit
 ) {
     LazyColumn(modifier) {
-        items(movies /* key = Movie::id */) { movie ->
-            if (movie == null) return@items
-            TrendingCard(movie, block)
+        items(count = movies.itemCount) {
+            val item = movies[it] ?: return@items
```
Comment (Nanamare) [inline app/src/main/java/com/nanamare/movie/ui/screen/TrendingScreen.kt:62]: p5 : item -> movie ?!

File: gradle.properties [hunk 3]
```diff
@@ -22,4 +22,6 @@ kotlin.code.style=official
 # Enables namespacing of each library's R class so that its R class includes only the
 # resources declared in the library itself and none from the library's dependencies,
 # thereby reducing the size of the R class for that library
-android.nonTransitiveRClass=true
\ No newline at end of file
+android.nonTransitiveRClass=true
+android.defaults.buildfeatures.buildconfig=true
+android.nonFinalResIds=false
```
Comment (Nanamare) [inline gradle.properties:27]: @l2hyunwoo 
Thanks!
https://developer.android.com/build/releases/past-releases/agp-8-0-0-release-notes#default-changes

**Review by Nanamare — APPROVED** [review]
SGTM 🥇

Merge 🚀

**Comment by l2hyunwoo** [comment]
@Nanamare 이 PR shared만 추가해놔서 앞으로 프로젝트 모듈을 어떻게 분리해나갈건지도 생각해보는게 좋을 것 같아요. 저는 일단 드로이드나이츠에서 [승욱님이 제시한 방법](https://github.com/droidknights/DroidKnights2023_App/pull/261)을 활용하거나 [droidKaigi에서 활용하는 아키텍처](https://github.com/DroidKaigi/conference-app-2023)를 활용해볼까 합니다.

File: app/src/main/java/com/nanamare/movie/ui/screen/TrendingScreen.kt (hunk 2 above)
Comment (l2hyunwoo) [reply app/src/main/java/com/nanamare/movie/ui/screen/TrendingScreen.kt]: [수정](https://github.com/Nanamare/MovieCompose/pull/113/commits/3d9196c946b9792a224f6cd0843760ce235f1aec)되었습니다

File: gradle.properties (hunk 3 above)
Comment (l2hyunwoo) [reply gradle.properties]: 👍🏻

**Pull request merged**